linalg/solvers.hpp

```diff
@@ -169,10 +169,13 @@ class IterativeSolver : public Solver
 
    ///@}
 
-   /// Return the dot product of @a x and @a y
-   double Dot(const Vector &x, const Vector &y) const;
+   /** @brief Return the standard (L2) inner dot product of @a x and @a y
+       @details Overriding this method in a derived class enables a 
+       custom inner product.
+      */           
+   virtual double Dot(const Vector &x, const Vector &y) const;
 
-   /// Return the 2-norm of @a x
+   /// Return the inner product norm of @a x
```

Comment:
Maybe clarify: `Return the inner product norm of @a x, with the inner product defined by Dot()`.